ts/utils/fetch.ts

```diff
@@ -90,24 +78,16 @@ export function retryLogicForTransientResponseError(
 
 /**
  * This is a fetch with timeouts, constant backoff and with the logic
- * that handles 404s as transient errors, this "fetch" must be passed to
- * createFetchRequestForApi when creating "getPaymentId"
+ * that handles 404s as transient errors.
  */
 export const constantPollingFetch = (
   shouldAbort: Promise<boolean>,
   retries: number,
   delay: number,
   timeout: Millisecond = fetchTimeout
```

Comment:
After a brief chat with @fabriziofff it turns out that the timeout parameter applies only _per request_. Will update accordingly.